iOS/MobileStarter/react-app/src/HubScreen.tsx

```diff
@@ -239,247 +114,109 @@ export function HubScreen(props: HubScreenProps) {
-        deleteButton = (
-          <div className="delete-button" onClick={async (e) => {
-            e.stopPropagation();
-            try {
-              await MobileCore.deleteCachedBriefcase(briefcase);
-              if (!isMountedRef.current) return;
-              if (project) {
-                selectProject(project);
-              }
-            } catch (error) {
-              // There was a problem deleting the cached briefcase. Show the error.
-              presentError("DeleteErrorFormat", error, "HubScreen");
-            }
-          }}>
-            <IconImage iconSpec="icon-delete" />
-          </div>
-        );
-      }
-      return (
-        <div className="imodel-row" key={index}>
-          <Button
-            onClick={() => handleSelectIModel(index)}
-            title={getTitle(title, briefcase)}
-          >
-            {deleteButton}
-          </Button>
-        </div>);
-    } else {
-      return <Button
-        key={index}
-        onClick={async () => {
-          if (title === signOutLabel) {
-            if (IModelApp.authorizationClient instanceof ITMAuthorizationClient) {
+  }, [isMountedRef, project, initialized]);
+
+  let moreButton: React.ReactNode;
+  let stepContent: React.ReactNode;
+
+  switch (hubStep) {
+    case HubStep.SelectProject:
+      stepContent = <ProjectPicker signedIn={initialized}
+        onSelect={(newProject) => {
+          saveActiveProject(newProject);
+          setProject(newProject);
+          setHubStep(HubStep.SelectIModel);
+        }}
+        onError={() => setHubStep(HubStep.Error)}
+      />;
+      break;
+
+    case HubStep.SelectIModel:
+      const actions: AlertAction[] = [];
+      if (haveCachedBriefcase) {
+        actions.push({
+          name: "deleteAll",
+          title: deleteAllDownloadsLabel,
+          onSelected: async () => {
+            if (project) {
               try {
-                await IModelApp.authorizationClient.signOut();
+                await MobileCore.deleteCachedBriefcases(project.id);
+                if (!isMountedRef.current) return;
+                setHaveCachedBriefcase(false);
               } catch (error) {
-                console.log(`Error signing out: ${error}`);
+                // There was a problem deleting the cached briefcases. Show the error, then refresh
+                // anyway so if any were successfully deleted, it will reflect that.
+                presentError("DeleteAllErrorFormat", error, "HubScreen");
+                setProject({ ...project });
               }
             }
-            onBack();
-          }
-          if (hubStep === HubStep.SelectProject) {
-            handleSelectProject(index, title);
           }
+        });
+      }
+      if (project) {
+        actions.push({
+          name: "changeProject",
+          title: changeProjectLabel,
+          onSelected: () => setHubStep(HubStep.SelectProject),
+        })
+      }
+      if (actions.length > 0) {
```

Comment:
Note that if we assume that project is defined, actions.length will always be > 0.

Reply:
Fixed